### DISCUSSION TITLE
Why the _ (private?) tasks are not allowed to be run from outside?

### DISCUSSION THREAD

**Discussion by mgzenitech** [nat-n/poethepoet#204] [Q&A]
I'm curious, because I have a case where I need to hide poe task from `poetry poe --help` but still be able to run it in part of automated calls. Maybe there are other ways to hide the tasks then?

**Comment by nat-n** [comment]
Indeed the thinking behind hidden tasks today assumes hiding a task from the docs and making it really inaccessible always go together, because a hidden task exists purely for internal usage.

It also provides a nice way to segregate the few special tasks poe has like `poe _zsh_completion`.

Could you tell me more about what kind of task you want to hide from docs but still be able to call it directly from a script? do you have many of these tasks? Maybe we could think of a workaround or elegant feature to enable this?

**Reply by mgzenitech** [reply]
So the task I had to create is basically `merge_files` which is triggered automatically by VSCode when developer edits one of few possible project files and final result is `built` configuration files from multiple sources.

Maybe it would be possible to relax the restriction of not being able to run `_` tasks? Or at least having the ability to trigger them some other way...

Here's excerpt from my config for one of multiple merge ops:
```
  "runOnSave.commands": [
    {
      "async": false,
      "command": "poetry poe merge -s ${workspaceFolder}/.vscode/settings.*.json -d ${workspaceFolder}/.vscode/settings.json -t json",
      "globMatch": "${workspaceFolder}/.vscode/settings.*.json",
      "runIn": "backend"
    },
    ...
]
```

**Reply by nat-n** [reply]
Right so the basic scenario is that you have tasks that you want to run from another tool, but you still don't want them to be user facing/documented. I'll have a think about this.
